### DISCUSSION TITLE
[Feature] API call to move asset to External Library

### DISCUSSION THREAD

**Discussion by psyciknz** [immich-app/immich#10092] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Not sure if it really just a usecase of mine or if would be useful to others, but I'd really like the ability to make an API call to move an asset to a path in an external library.  This would of course require write access to that library.

I foresee something like a move function with an asset id and a destination path that would be visible to the microservices (I assume that is what would perform the action).

Or even a copy to external location, and a delete library asset would be just as welcome instead of a move.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
I think this is a duplicate request, but can't find the other one... For now, some related ones: #5453, #1992

**Reply by AlexSamad** [reply]
I wrote a separate issues - mine was slightly different but some way close to similiar

main difference was it moves from ext to int library - doesn't ask you to break current rules about ext libraries - I've copied here for history 


So just to describe my understanding
internal library - immich has full control of the image - no external interference is allowed - so no cli
external library - immich has only read only on the image - it can if the space has RW add xmp files to it

My work flow for importing lots of files has been to create an external library - because I am usually working on the space from another app / pc or ??? For example batch scanning in old photo's

What I would like to do is look at the photos in the immich library (so the main page - which is a mixture of ext and int libraries)
I would also like to look at via the folder view - I can group my photo's here by scan date.
But once I am i okay with all of the work I am going to do I would like to move / migrate the image into the int library - immich should allow this to happen - now to keep with the do not touch images in ext library I think duplicate the file into int lib, than write a new file into the original directory with extension something like -imported.. The duplicate function would assume is in immich right now and immich already has a function for moving file around that could be used to place the duplicate !

The only other work is when reading a ext library - you need to ignore any file which has -imported

so example if ext library has
TodaysFile.jpg

then after importing it would look like
TodaysFile.jpg
TodaysFile.jpg-import
TodaysFile.jpg-xmp (<< maybe depending on what work has been done in immich)

This would allow the user to use cli tool to batch find and delete these image files that have been processed
